helix-core/src/main/java/org/apache/helix/controller/rebalancer/DelayedAutoRebalancer.java

```diff
@@ -363,6 +364,9 @@ protected Map<String, String> computeBestPossibleStateForPartition(Set<String> l
     currentMapWithPreferenceList.keySet().retainAll(preferenceList);
 
     combinedPreferenceList.addAll(currentInstances);
+    if (cache != null) {
+      combinedPreferenceList.removeAll(cache.getUnknownInstances());
+    }
```

Comment:
I removed this and now pass in a current state that excludes UNKNOWN instances. This current state is generated in CurrentStateComputationStage and is added as another cluster event attribute called CURRENT_STATE_EXCLUDING_UNKNOWN. This will ensure that there is not n -> n+1 for UNKNOWN InstanceOperation.

I have also changed MessageGenerationStage to drop replicas that are in current state and not in BestPossibleState instanceStateMaps so that we will be able to drop when setting to UNKNOWN.